src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -88,29 +87,29 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         }
 
         if (null !== $dateTimeFormat) {
-            $object = \DateTime::class === $type ? \DateTime::createFromFormat($dateTimeFormat, $data, $timezone) : \DateTimeImmutable::createFromFormat($dateTimeFormat, $data, $timezone);
```

Comment:
As `$type` is an argument, be careful to still support this way for users to decide of final normalization.